### ISSUE TITLE
Flow triggering for Tag Set event not working...

### ISSUE THREAD

**Issue by madscientist42** [Raudius/files_scripts#119]
Version used: 3.0.0

Nextcloud vrsion:  27.1.2

Action : Set a tag on a file that is supposed to be monitored for in the Workflow engine.

Expected: It triggers the desired file action script.

What's happening: Nada.  No execution of any of the File Action scripts on the event.  Know it's not Nextcloud because the External Scripting app does the right things, even though that's not overly helpful here.  Needed what this app offered instead.  The script, when it's launched from the menus against a given file works with the functionality of the script as intended, so it's not the script either.  It just sits there and never seems to act on the tag hook.

**Comment by Raudius** [comment]
I was not able to reproduce this, could you provide some more details on how you set up the flow?

For reference this is what I tried
![image](https://github.com/Raudius/files_scripts/assets/3178979/b01910c4-afe3-450c-8e42-50bb4d8f33d5)

In the test script I simply did a `log` call to check that it got triggered

```lua
log("test")
```

**Comment by madscientist42** [comment]
Give me a bit.  Had to set aside the project for my main client's current
project.  More than happy help you help me diagnose this.

**Comment by madscientist42** [comment]
Well, I finally got a bit of info.  Area that I'm trying to make it work is in a group folder.  If you run the Flow with your plugin, it works as expected on the tagging event when it's in the personal directory and fails miserably when it's in the group folder.  Will dig a bit more to see if it's something that I can reproduce consistently there.

**Comment by madscientist42** [comment]
Yeah, it very much looks like it's a group folder related issue.  Nothing else.  Files in my personal area work right.  Files int he group share...don't.

**Comment by Raudius** [comment]
Great, thanks for taking the time. I will test it later and try to fix it.

**Comment by madscientist42** [comment]
No problems.  Like I said, more than happy to help.  :D

**Comment by Raudius** [comment]
Hmm, after further testing I still cannot reproduce (tested on NC27 and NC28)

I tested the `groupfolders` app and even shared folders and external storage. They all worked for me.

Could you provide more details? 

- How have you set up the flow?
- What permissions do you have on the groupfolder?
- Can you try adding something like `log("triggered action")` at the top of the script to check that its not being triggered?
- Is there anything relevant in the `nextcloud.log` when the action should have triggered?

**Comment by madscientist42** [comment]
Sure.  Sorry about the delay.  Like a one-legged man in a butt-kickin' contest right now.  X-D

Flow is set up as a second flow against your plugin.  First one generates an upload against a new file in an already specially tagged directory.  (Trying to set up an approval pipeline for a Graphic Novel being published over time- NextCloud is MUCH easier to explain than any aspect of WordPress...  X-D)   That works fine.  The "competing" plugin works fine on this- but is weak in comparison and flatly can't do what I need of it.  X-D

Permissions should be R/W for all group members and I am admin.

Already done in the case of my side of things- it's how I know it works fine in non-groups at least on my setup.

Please define, "relevant"- no errors and no log entries from the lua script.    The one thing that might be relevant is that I've had issues getting the Lua PHP plugin to work right with your script.  I AM using the experimental interpreter with the intent of moving over once I sort out this problem (Fix one thing at a time...less to break...  X-D)   When I've tried to run much of anything in the real Lua environment (Which would've been nice- I could've done this more properly in all-Lua code instead of a cURL kludge...)  it just quietly completes and does nothing.  

It's my box, so I can control any aspects or try things.  My availability will be sporadic for another week to three at the least.  Client's stuff...family obligations for Mother and Wife (Both in rehab right now...) have taken a few priority interrupts.  Still serious- this has epic potential and I am more than happy to help.  This can be the swiss army knife/chainsaw of Flows and it can make real magic happen.

**Comment by madscientist42** [comment]
Later this evening I can try a few things and give you maybe a bigger data dump to help.  If not, I'll aim for tomorrow my time.  For me, it's not TOO pressing, but the Business Partner, the gent with the Novel we're turning into GN for, is itching to get it live for the new year if I can or get it up and limping enough to count.

**Comment by madscientist42** [comment]
(That being said, he wants it **_RIGHT_** so we don't muck with the workflow too much over time.  X-D)

**Comment by Raudius** [comment]
Sorry to hear you are going through a stressful time, I can fully relate with the feeling.

No rush to get back but might also help to know the `groupfolders` version

> Please define, "relevant"- no errors and no log entries from the lua script. 

Was mostly curious if there was an error log with a trace leading back to the `files_scripts` app

> I AM using the experimental interpreter 

This should not be a problem for this issue. I just dont recommend using it because its an interpreter I wrote to run on PHP and does not implement the full Lua spec. It also doesnt have the stability or time-hardening that the plugin interpreter has.

**Comment by madscientist42** [comment]
No logs there.  Will get you the groupfolders version shortly.

Figured it wasn't good.  Was trying to get something up quick and then "fix" it fully later.  I really do want to do this via Lua proper- it will run faster, etc. and should let me have the full power of Lua.

**Comment by Raudius** [comment]
@madscientist42 Just reaching out to see if you managed to fix it or if you made any progress at all on this issue?

I also remembered that the workflow engine has its own log file `flow.log` in the nextcloud data folder, you could try checking in there. Possibly also lower your log level in the `config.php` to get more verbose output.

**Comment by madscientist42** [comment]
No progress...still trying but up to my backside with alligators.  Will
look in the flow.log shortly.

**Comment by ZeiTee** [comment]
I got the same issue while trying to setup a approval workflow with a flow operation executing a file script on approval (using the tag assigned event).
I discovered two issues here. 
The first is the one resolved by my pull request. Here the node in the groupfolder is received from the root folder instead of the user root folder which can lead to a node path in a other users directory if both users have the same groupfolder as mountpoint in the database.
The second one is connected to the approval app where the approved tag is added before the pendig tag is removed which in my case (I have a check „!is“ ->  „pending tag“ in the operation triggering the file script) don‘t trigger the file script flow.
